src/Functions/getSetting.cpp

```diff
@@ -19,20 +19,23 @@ namespace ErrorCodes
 namespace
 {
 
+enum class ErrorHandling : uint8_t
+{
+    Exception,
+    Default,
+};
+
 /// Get the value of a setting.
 class FunctionGetSetting : public IFunction, WithContext
 {
 public:
-    static constexpr auto name = "getSetting";
-
-    static FunctionPtr create(ContextPtr context_) { return std::make_shared<FunctionGetSetting>(context_); }
-    explicit FunctionGetSetting(ContextPtr context_) : WithContext(context_) {}
+    explicit FunctionGetSetting(ContextPtr context_, String name_, ErrorHandling error_handling_) : WithContext(context_), name(name_), error_handling(error_handling_) {}
```

Comment:
I'd suggest to make `FunctionGetSetting` a template with `template <ErrorHandlingMode mode>`, which can be used with `constexpr`. So there's no need to have additional parameters in constructor

Reply:
Done, thanks! I checked a couple of other similar implementations in Functions/ and used that style.